pilot/pkg/networking/core/v1alpha3/configgen.go

```diff
@@ -78,5 +78,10 @@ func (configgen *ConfigGeneratorImpl) CanUsePrecomputedCDS(proxy *model.Proxy) b
 		return false
 	}
 
+	// If proxy locality set, support locality weighted load balancing
+	if proxy.Locality != "" {
+		return false
```

Comment:
Sigh.. this will kill all the caching benefits

Reply:
@mandarjog could you quantify the perf if we disable the CDS cache? If there is not much of a regression, then we can drop the CDS cache in favor of locality routing which more people want.

Reply:
@hzxuzhonghu for the moment, I suggest you go with this model, where if locality is set, you disable the cache. if @mandarjog says the caching has little benefit, then we can get rid of CDS cache completely and will simplify the code here. In terms of the EDS code, if there is any cache there, we need to partition that cache by localities so that given a proxy, we get the proxy's locality and get the EDS cache for that locality, and then do the rest of EDS stuff.

Reply:
ok

Reply:
I can further optimize this.   Not  modify `CanUsePrecomputedCDS`.  And when we can use precomputed Clusters, just apply locality weighted LB setting. This can  maximize the use of caching.

Reply:
done